internal/testutil/harnessutil/harnessutil.go

```diff
@@ -355,22 +356,29 @@ type cachedCompilerHost struct {
 
 var sourceFileCache sync.Map
 
-func (h *cachedCompilerHost) GetSourceFile(fileName string, path tspath.Path, languageVersion core.ScriptTarget) *ast.SourceFile {
+func (h *cachedCompilerHost) GetSourceFile(fileName string, path tspath.Path, languageVersion core.ScriptTarget, packageJsonScope *packagejson.InfoCacheEntry) *ast.SourceFile {
 	text, _ := h.FS().ReadFile(fileName)
 
 	type sourceFileCacheKey struct {
 		core.SourceFileAffectingCompilerOptions
-		fileName        string
-		path            tspath.Path
-		languageVersion core.ScriptTarget
-		text            string
+		fileName             string
+		path                 tspath.Path
+		languageVersion      core.ScriptTarget
+		packageJsonTypeValue string
+		text                 string
+	}
+
+	packageJsonTypeValue := ""
+	if packageJsonScope != nil {
+		packageJsonTypeValue = packageJsonScope.Contents.Type.Value
```

Comment:
I'm noticing that only `Type.Value` is ever used. Do we need to store a full `InfoCacheEntry` on each file? Would storing just the type as a string be sufficient?

As it stands right now, this is _technically_ unsafe because we're reusing ASTs where the entries may be entirely different objects. But the fact that only this string matters does seem to imply it's possible.